From ef0ec7db74399fb880887a6e9954ffa6db788b8c Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 22 Nov 2016 07:34:45 -0500
Subject: [PATCH 11/24] always use the WebView sandbox

---
 .../WebViewChromiumFactoryProvider.java         | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/android_webview/glue/java/src/com/android/webview/chromium/WebViewChromiumFactoryProvider.java b/android_webview/glue/java/src/com/android/webview/chromium/WebViewChromiumFactoryProvider.java
index e808a822dca3..4da7ae31e0e9 100644
--- a/android_webview/glue/java/src/com/android/webview/chromium/WebViewChromiumFactoryProvider.java
+++ b/android_webview/glue/java/src/com/android/webview/chromium/WebViewChromiumFactoryProvider.java
@@ -275,21 +275,8 @@ public class WebViewChromiumFactoryProvider implements WebViewFactoryProvider {
                 CommandLineUtil.initCommandLine();
             }
 
-            boolean multiProcess = false;
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                // Ask the system if multiprocess should be enabled on O+.
-                multiProcess = webViewDelegate.isMultiProcessEnabled();
-            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-                // Check the multiprocess developer setting directly on N.
-                multiProcess = Settings.Global.getInt(
-                                       ContextUtils.getApplicationContext().getContentResolver(),
-                                       Settings.Global.WEBVIEW_MULTIPROCESS, 0)
-                        == 1;
-            }
-            if (multiProcess) {
-                CommandLine cl = CommandLine.getInstance();
-                cl.appendSwitch(AwSwitches.WEBVIEW_SANDBOXED_RENDERER);
-            }
+            CommandLine cl = CommandLine.getInstance();
+            cl.appendSwitch(AwSwitches.WEBVIEW_SANDBOXED_RENDERER);
 
             int applicationFlags = ContextUtils.getApplicationContext().getApplicationInfo().flags;
             boolean isAppDebuggable = (applicationFlags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
-- 
2.23.0

