From 6d485346c05019a2f7224db31613accfdea77525 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 22 Dec 2016 07:15:34 -0500
Subject: [PATCH 16/24] enable -fwrapv in non-UBSan builds for clang

---
 build/config/compiler/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 113d4031d797..16bb822bdb55 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -271,7 +271,13 @@ config("compiler") {
       # crbug.com/485542
       ldflags += [ "-Wl,--fatal-warnings" ]
     }
+    if (is_clang && !is_ubsan && !is_ubsan_security) {
+        cflags += [ "-fwrapv" ]
+    }
   } else {
+    if (is_clang && !is_ubsan && !is_ubsan_security) {
+        cflags += [ "-fwrapv" ]
+    }
   }
 
   if (is_clang && is_debug) {
-- 
2.23.0

