From 668a7cbb158b8b1135f4a49ea8753cd53a3b856a Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 22 Dec 2016 07:15:34 -0500
Subject: [PATCH 17/23] enable -fwrapv in non-UBSan builds for clang

---
 build/config/compiler/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 70375bd88c41..0da9c137adec 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -277,6 +277,9 @@ config("compiler") {
       # crbug.com/485542
       ldflags += [ "-Wl,--fatal-warnings" ]
     }
+    if (is_clang && !is_ubsan && !is_ubsan_security) {
+        cflags += [ "-fwrapv" ]
+    }
   } else {
     cflags += [
       # Assume UTF-8 by default to avoid code page dependencies.
@@ -286,6 +289,9 @@ config("compiler") {
       # Don't look for includes in %INCLUDE%.
       cflags += [ "/X" ]
     }
+    if (is_clang && !is_ubsan && !is_ubsan_security) {
+        cflags += [ "-fwrapv" ]
+    }
   }
 
   # Eliminate build metadata (__DATE__, __TIME__ and __TIMESTAMP__) for
-- 
2.14.3

